agent/consul/fsm/commands_oss.go

```diff
@@ -461,12 +462,12 @@ func (c *FSM) applyConnectCAOperation(buf []byte, index uint64) interface{} {
 		if !act {
 			return act
 		}
-
-		act, err = c.state.CACheckAndSetConfig(index+1, req.Config.ModifyIndex, req.Config)
+		var emptyResp interface{}
+		err = c.state.CACheckAndSetConfig(index+1, req.Config.ModifyIndex, req.Config)
```

Comment:
I believe this `index+1` is also a bug. It should be just `index`. We could fix it here at the same time, but also fine to leave it for a separate PR.

Reply:
I was thinking of fixing this in the upstream branch so we can backport it.